static/app/components/events/viewHierarchy/tree.tsx

```diff
@@ -0,0 +1,114 @@
+import {ReactNode, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {IconAdd, IconSubtract} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import space from 'sentry/styles/space';
+
+type NodeProps = {
+  type: string;
+  children?: ReactNode[];
+  identifier?: string;
+};
+
+function Node({type, identifier, children}: NodeProps) {
+  const [isExpanded, setIsExpanded] = useState(true);
+  return (
+    <NodeContents>
+      {children?.length && (
+        <IconWrapper
+          aria-label={isExpanded ? t('Collapse') : t('Expand')}
+          isExpanded={isExpanded}
+          onClick={evt => {
+            evt.preventDefault();
+            setIsExpanded(!isExpanded);
+          }}
+        >
+          {isExpanded ? (
+            <IconSubtract size="9px" color="white" />
+          ) : (
+            <IconAdd size="9px" color="white" />
+          )}
+        </IconWrapper>
+      )}
+      <NodeTitle>{identifier ? `${type} - ${identifier}` : type}</NodeTitle>
+      {isExpanded && children}
+    </NodeContents>
+  );
+}
+
+function Tree({hierarchy}) {
```

Comment:
No types?

Reply:
Yeah, I forgot the types so I put this PR on draft mode until I could get to them tomorrow 😅  I appreciate you taking a look though

Reply:
Ah sorry. Throw me on review once it's ready to go :)

Reply:
Going to resolve this and let you go for round 2